### PR TITLE
Fix cache policy calculation

### PR DESCRIPTION
**Pull request by fabsrc**
Fixes apollographql/apollo-cache-control#11

TODO:

* [ ] Update CHANGELOG.md with your change (include reference to issue & this PR)
* [x] Make sure all of the significant new logic is covered by tests
* [x] Rebase your changes on master so that they can be merged easily
* [x] Make sure all tests and linter rules pass

### CODE REVIEW

**Comment by glasser** [comment]
Awesome - I am working on some cache related features this week and had noticed the same concern. Will probably merge Monday!

File: packages/apollo-cache-control/src/index.ts [hunk 1]
```diff
@@ -156,7 +156,7 @@ export class CacheControlExtension<TContext = any>
     let scope: CacheScope = CacheScope.Public;
 
     for (const hint of this.hints.values()) {
-      if (hint.maxAge) {
+      if (hint.maxAge !== undefined) {
         lowestMaxAge = lowestMaxAge
```
Comment (glasser) [inline packages/apollo-cache-control/src/index.ts]: Should the check here (before the `?`) also be against `undefined`?

Comment (fabsrc) [reply packages/apollo-cache-control/src/index.ts]: Yes, you are right. Did check that case. 😕 
Just fixed it and adjusted the test.

**Comment by glasser** [comment]
Thanks!

**Pull request merged**